samples/bluetooth/direction_finding_central/src/main.c

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <sys/printk.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/direction.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include <sys/byteorder.h>
+
+/* Latency set to zero, to enforce PDU exchange every connection event */
+#define CONN_LATENCY 5U
+/* Arbitrary selected timeout value */
+#define CONN_TIMEOUT 400U
+/* Inteval used to run CTE request procedure periodically.
+ * Value is a number of connection events.
+ */
+#define CTE_REQ_INTERVAL (CONN_LATENCY + 10U)
+/* Length of CTE in unit of 8 us */
+#define CTE_LEN (0x14U)
+
+#define DF_FEAT_ENABLED BIT64(BT_LE_FEAT_BIT_CONN_CTE_RESP)
+
+static struct bt_conn *default_conn;
+static const struct bt_le_conn_param conn_params = BT_LE_CONN_PARAM_INIT(
+	BT_GAP_INIT_CONN_INT_MIN, BT_GAP_INIT_CONN_INT_MAX, CONN_LATENCY, CONN_TIMEOUT);
+
+#if defined(CONFIG_BT_CTLR_DF_ANT_SWITCH_RX)
+static const uint8_t ant_patterns[] = { 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xA };
+#endif /* CONFIG_BT_CTLR_DF_ANT_SWITCH_RX */
+
+static void start_scan(void);
+
+static const char *cte_type2str(uint8_t type)
+{
+	switch (type) {
+	case BT_DF_CTE_TYPE_AOA:
+		return "AOA";
+	case BT_DF_CTE_TYPE_AOD_1US:
+		return "AOD 1 [us]";
+	case BT_DF_CTE_TYPE_AOD_2US:
+		return "AOD 2 [us]";
+	default:
+		return "Unknown";
+	}
+}
+
+static const char *packet_status2str(uint8_t status)
+{
+	switch (status) {
+	case BT_DF_CTE_CRC_OK:
+		return "CRC OK";
+	case BT_DF_CTE_CRC_ERR_CTE_BASED_TIME:
+		return "CRC not OK, CTE Info OK";
+	case BT_DF_CTE_CRC_ERR_CTE_BASED_OTHER:
+		return "CRC not OK, Sampled other way";
+	case BT_DF_CTE_INSUFFICIENT_RESOURCES:
+		return "No resources";
+	default:
+		return "Unknown";
+	}
+}
```

Comment:
This could perhaps be moved to `direction.h` - Possible just as `inline` functions. It isn't something we've done a lot in the past, but possible a useful feature to have.

Reply:
I can do it but in separate PR. Then I'll update all samples that use similar code.